lib/sinatra/base.rb

```diff
@@ -1924,6 +1924,11 @@ class #{self.class}
   # inherit all settings, routes, handlers, and error pages from the
   # top-level. Subclassing Sinatra::Base is highly recommended for
   # modular applications.
+
+  # We need ruby2_keywords to make delegation work correctly on
+  # ruby2.7 and ruby3
+  require 'ruby2_keywords'
```

Comment:
I would put the require at the top of the file, somewhere at https://github.com/sinatra/sinatra/blob/cf404526373298f06a59e1a129b6cdf8282ae216/lib/sinatra/base.rb#L4-L10